### DISCUSSION TITLE
'__VLS_ctx.$t' is of type 'unknown'. in vue-i18n.

### DISCUSSION THREAD

**Discussion by maykeloo** [nuxt/nuxt#21843] [Questions]
Setup:
```
"@vueuse/core": "^10.0.2",
"vue-i18n": "^9.3.0-beta.17"
"nuxt": "^3.6.1",
```

From the beggining.

I am taking over a post-migration Nuxt project (2.15 -> 3+) in which each component is imported manually. I wanted to change this, but my components that were auto-imported were not recognised by Nuxt. They were getting the error 
`(property) ComponentName: unknown`, and were not being highlighted. The problem was this `"types"` property in `ts.config`.
`"types": ["@nuxt/types", "@nuxtjs/i18n"]`. 

The project is configured with vue-i18n:
```
import { createI18n, I18nOptions } from 'vue-i18n'

export default defineNuxtPlugin(({ vueApp }) => {
  const i18n = createI18n({
    legacy: false,
    globalInjection: true,
    locale: 'pl',
    messages: {
      pl: PL,
    },
    defaultLocale: 'pl',
    fallbackLocale: 'pl',
    datetimeFormats,
    numberFormats,
  })

  vueApp.use(i18n)
})
```
But as you can see, there is `"@nuxtjs/i18n"` in `"types"` in `ts.config` which is also implemented in the project, but it is not used anywhere, except that we have its types in `"types"`. 

When I remove the `"types"` property, the auto-import of the components works correctly, they highlight, its props are displayed, etc. And that is OK, that is what I wanted.

BUT...

When I uninstalled`"@nuxtjs/i18n"` which is not used anyway, my `$t` in `<template>` throws the error `'__VLS_ctx.$t' is of type 'unknown'.ts(18046)`. So the only problem remains `vue-i18n` which is implemented according to [Vue-i18n - Nuxt 3](https://vue-i18n.intlify.dev/guide/integrations/nuxt3.html)

**Comment by Proxx07** [comment]
How did you solve it? I have the same bug

**Comment by AzotoN1** [comment]
I have same issue

**Reply by ZadorozhnyiYevhenii** [reply]
I had the same issue and the following VS code setting solved it, set the vscode setting  "vue.server.hybridMode": true [automatically enable Hybrid Mode](https://github.com/vuejs/language-tools/pull/4206), [Hybrid Mode is disabled automatically because there is a potentially incompatible GitHub.copilot-chat TypeScript plugin installed](url)

**Reply by luke-z** [reply]
@ZadorozhnyiYevhenii Thanks, works for me!

**Comment by TheDogHusky** [comment]
Hey!
I have the same issue on webstorm: installed nuxt-i18n, and got the same type error.
Any ideas?

**Reply by neoighodaro** [reply]
Did you find a way around it?

**Reply by TheDogHusky** [reply]
Sadly no.. I still have no type help from my IDE with i18n, even though all the other parts of the library are typed correctly (such as the useI18n() composable)

**Reply by seho-nl** [reply]
I'm sadly enough experiencing the same issue in PHPStorm.

**Comment by abdallhsamy** [comment]
a simple solution is to create `types.d.ts` in project root and define `$t` 

```ts
declare module "@vue/runtime-core" {
    export interface ComponentCustomProperties {
        $t: (key: string, ...args: unknown[]) => string;
    }
}

```

**Comment by FallingEpiphany** [comment]
shouldnt the pack manage the type inference and the legal accessibility of its functions itself?😥

**Reply by TheDogHusky** [reply]
It should..
Best we can do it do it ourselves, and probably make a PR on the package for it.
